src/Hl7.Fhir.Core/Rest/Http/EntryToHttpExtensions.cs

```diff
@@ -0,0 +1,116 @@
+            else if (interactionType == TransactionBuilder.InteractionType.Search && handlingPreference != null)
+                request.Headers.TryAddWithoutValidation("Prefer", "handling=" + PrimitiveTypeConverter.ConvertTo<string>(handlingPreference));
+
+            if (entry.Resource != null)
+                setBodyAndContentType(request, entry.Resource, format, CompressRequestBody);
+
+            return request;
+        }
+
+        /// <summary>
+        /// Converts bundle http verb to corresponding <see cref="HttpMethod"/>.
+        /// </summary>
+        /// <param name="verb"><see cref="Bundle.HttpVerb"/> specified by input bundle.</param>
+        /// <returns><see cref="HttpMethod"/> corresponding to verb specified in input bundle.</returns>
+        private static HttpMethod getMethod(Bundle.HTTPVerb? verb)
+        {
+            switch(verb)
+            {
+                case Bundle.HTTPVerb.GET:
+                    return HttpMethod.Get;
+                case Bundle.HTTPVerb.POST:
+                    return HttpMethod.Post;
+                case Bundle.HTTPVerb.PUT:
+                    return HttpMethod.Put;
+                case Bundle.HTTPVerb.DELETE:
+                    return HttpMethod.Delete;
+            }
+            throw new HttpRequestException($"Valid HttpVerb could not be found for verb type: [{verb}]");
+        }
+
+        private static void setBodyAndContentType(HttpRequestMessage request, Resource data, ResourceFormat format, bool CompressRequestBody)
+        {
+            if (data == null) throw Error.ArgumentNull(nameof(data));
+
+            byte[] body;
+            string contentType;
+
+            if (data is Binary bin)
+            {
+                body = bin.Content;
+                // This is done by the caller after the OnBeforeRequest is called so that other properties
+                // can be set before the content is committed
+                // request.WriteBody(CompressRequestBody, bin.Content);
+                contentType = bin.ContentType;
+            }
+            else
+            {
+                body = format == ResourceFormat.Xml ?
+                    new FhirXmlSerializer().SerializeToBytes(data, summary: Fhir.Rest.SummaryType.False) :
+                    new FhirJsonSerializer().SerializeToBytes(data, summary: Fhir.Rest.SummaryType.False);
+
+                // This is done by the caller after the OnBeforeRequest is called so that other properties
+                // can be set before the content is committed
+                // request.WriteBody(CompressRequestBody, body);
+                contentType = Hl7.Fhir.Rest.ContentType.BuildContentType(format, forBundle: false);
+            }
+
+            request.Content = new ByteArrayContent(body);
+
+            // MediaTypeHeaderValue cannot accept a content type that contains charset at the end, so that value must be split out.
```

Comment:
Yes, I noticed the same, weird. Maybe I'll take a look at the reference source to see why.

Reply:
If you look at https://github.com/dotnet/corefx/blob/master/src/System.Net.Http/src/System/Net/Http/Headers/MediaTypeHeaderValue.cs#L203, it seems there is support for parsing mediatypes with parameters...

Reply:
Will replace with `MediaTypeHeaderValue.Parse(contentType)`